doc/source/data/data-internals.rst

```diff
@@ -92,6 +92,44 @@ To change the size at which Ray Data splits blocks, configure
 Ray Data can’t split rows. So, if your dataset contains large rows (for example, large
 images), then Ray Data can’t bound the block size.
 
+
+Shuffle Algorithms
+------------------
+
+Ray Data implements two main shuffle algorithms that power different operations:
+
+.. _hash-shuffle:
+
+Hash-shuffling
+~~~~~~~~~~~~~
+
+.. note:: Hash-shuffling is available in Ray 2.46
+
+Hash-shuffling is a classical hash-partitioning based shuffling where:
+
+1. **Partition phase:** rows in every block are hash-partitioned based on values in the *key columns* into a specified number of partitions, following a simple residual formula of ``hash(key-values) % N`` (used in hash-tables and pretty much everywhere).
+2. **Push phase:** partition's shards from individual blocks are then pushed into corresponding aggregating actors (called ``HashShuffleAggregator``s) handling respective partitions.
+3. **Reduce phase:** aggregators combine received individual partition's shards back into blocks optionally applying additional transformations before producing the resulting blocks.
+
+Hash-shuffling is particularly useful for operations that require deterministic partitioning based on keys, such as joins, group-by operations, and key-based repartitioning, by
+ensuring that rows with the same key-values are being placed into the same partition.
+
+.. note:: To use hash-shuffling in your aggregations and repartitioning operations, you need to currently specify
+    ``ray.data.DataContext.get_current().shuffle_strategy = ShuffleStrategy.HASH_SHUFFLE`` before invoking corresponding ``Dataset`` operations.
```

Comment:
```suggestion
    ``ray.data.DataContext.get_current().shuffle_strategy = ShuffleStrategy.HASH_SHUFFLE`` before creating a ``Dataset``. 
```

Reply:
That's not necessarily the case -- they need to set it before they invoke the operations. It could be the case that they have to fine-tune settings for one op, and then again differently for the other (not shuffle strategy specifically)

Reply:
Oh, i stand corrected here. Let me fix this one